src/modules/InstURL.rb

```diff
@@ -86,24 +87,27 @@ def RewriteCDUrl(url)
     def installInf2Url(extra_dir = "")
       return @installInf2Url unless @installInf2Url.nil?
 
-      parts = URL.Parse(Linuxrc.InstallInf("ZyppRepoURL"))
+      zypp_repo_url = Linuxrc.InstallInf("ZyppRepoURL").to_s
 
-      if parts.empty?
+      if zypp_repo_url.empty?
         # Make it compatible with the current behaviour when
         # install.inf does not exist.
         log.warn "No URL specified through ZyppRepoURL"
-        parts = URL.Parse("cd:///")
+        zypp_repo_url = "cd:/"
       end
 
-      parts["path"] = File.join(parts["path"], extra_dir) unless extra_dir.empty?
+      uri = URI(zypp_repo_url)
```

Comment:
I think Martin's point is that we parse and then build the URL back. That might not produce the very same string in some edge cases. We should return the original string and do the modification *only* when really need (HTTPS *and* `ssl_verify=no` used).